### ISSUE TITLE
Files are not watched if we once unwatch it and then again want to watch it

### ISSUE THREAD

**Issue by Myasir92** [paulmillr/chokidar#1027]
When I add the files in watcher to watch it and if any changes occur then show us a message but when I unwatch any file and then again want to watch that files then it show message of successfully added but didn't watch that file if we change any in that file it didn't show the message

**Comment by paulmillr** [comment]
chok version? node version? os version?

**Comment by Myasir92** [comment]
Cholidar Version:3.4.2
Node version: v12.16.3
OS Version: Window 10

**Comment by paulmillr** [comment]
Can you create a test repo that reproduces the case? If this was common, it would have been reported before.

**Comment by SnejUgal** [comment]
I believe I'm hitting the same issue, so will describe my issue in detail. I have this watcher:

```const driveWatcher = chokidar
.watch([], { depth: 0, disableGlobbing: true })
.on(`addDir`, directory => {
  // ..
})
.on(`unlinkDir`, directory => {
  // ..
});
```

Whenever a drive is mounted, I call `driveWatcher.add(mountPoint)`, where `mountPoint` is e.g. `/media/user/foo`. When I add this path the first time, `addDir` is fired for `/media/user/foo` and also its subdirectories, e.g. `/media/user/foo/egg`. When the drive is unmounted, I call `driveWatcher.unwatch(mountPoint)`. But when the drive is mounted the second time and I call `driveWatcher.add(mountPoint)`, `addDir` is fired only for `/media/user/foo but not its subdirectories`, e.g. `/media/user/foo/egg`.

A reduced test case should be:

1. `mkdir -p foo/{bar,baz}`
2. Watch `foo`. `addDir` fires for `foo`, `foo/bar` and `foo/baz`
3. Unwatch `foo`
4. Watch `foo` again. `addDir` is expected to fire for `foo`, `foo/bar` and `foo/baz` like before, but it will only fire for `foo`

I guess the problem here is that `unwatch` doesn't remove the appropriate paths in `_watched`. E.g. this is what `console.log(driveWatcher)` shows after unwatching a drive:

```
FSWatcher {
  // ..
  _watched: Map(3) {
    '/media/snejugal' => DirEntry {
      path: '/media/snejugal',
      _removeWatcher: [Function: bound _remove],
      items: Set(0) {}
    },
    "/media/snejugal/SnejUgal's" => DirEntry {
      path: "/media/snejugal/SnejUgal's",
      _removeWatcher: [Function: bound _remove],
      items: [Set]
    },
    "/media/snejugal/SnejUgal's/Videos" => DirEntry {
      path: "/media/snejugal/SnejUgal's/Videos",
      _removeWatcher: [Function: bound _remove],
      items: Set(0) {}
    }
  },
  _closers: Map(0) {},
  _ignoredPaths: Set(2) {
    "/media/snejugal/SnejUgal's",
    "/media/snejugal/SnejUgal's/**"
  },
  // ..
}
```

If I do a dirty `driveWatcher._watched.clear()`, `addDir` is fired for all subdirectories again.

**Comment by absir1949** [comment]
@paulmillr hello.May I ask, has this problem been solved? i have the same problem on mac os  12.4

**Comment by paulmillr** [comment]
@absir1949 since the issue is not closed, the problem has not been solved.
